3rdParty/fuerte/src/AsioSockets.h

```diff
@@ -218,6 +238,16 @@ struct Socket<fuerte::SocketType::Ssl> {
           return canceled; 
         });
   }
+
+  void rearm() {
+    // move away old socket, in case it is still in use by some background operation
+    FUERTE_ASSERT(deadSockets.capacity() > deadSockets.size());
+    deadSockets.push_back(std::move(socket));
```

Comment:
Do we still need this?

Reply:
Probably not. I am tempted to remove it and see how things go. This would be the best solution.